### ISSUE TITLE
assembly failing

### ISSUE THREAD

**Issue by kalyanlanka** [neowit/tooling-force.com#9]
assembly is failing with NPEs

Here is the stack trace
````
> assembly
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list
[trace] Stack trace suppressed: run last test:test for the full output.
[trace] Stack trace suppressed: run last test:test for the full output.
[error] Could not run test com.neowit.apex.SoqlGrammar: java.lang.NullPointerException
[error] Could not run test com.neowit.apex.CodeCompletions: java.lang.NullPointerException
[info] MetaXmlTest:
OK
OK
OK
[info] StubFileGeneratorTest:
[info] - Object type name in Trigger detection 1
[info] - Object type name in Trigger detection 2 - with blank line
[info] - Object type name in Trigger detection 3 - with comments
[info] - Object type name in Trigger detection 4 - with long comments
[info] - Object type name in Trigger detection 5 - with long trigger body
[info] - Object type name in Trigger detection 6 - with multi-line definition
[info] - Object type name in Trigger detection 7 - single line body
[info] ConfigTest:
[info] - No Command Params
[info] - Key without value
[info] - Missing ConfigFile
[info] - Existing config, but missing parameter
[info] - Command Line - Key Without Value #1
[info] - Command Line - Key Without Value #2
[info] - Command Line - Key Without --
[info] - Command Line - value enclosed in quotes --
[info] - Parameter from command line take priority over config
[info] - Parameter from command line take priority over --config
[info] SoqlParserUtilsTest:
[info] - Find FROM in outer select (with nested select)
[info] - Find FROM in outer select (without nested select) #1
[info] - Find FROM in outer select (without nested select) #2
[info] - Find FROM in outer select (without nested select), without WHERE
[info] - Find FROM in outer select (without nested select) with incomplete WHERE
[info] - Find FROM in nested select
[info] - Find FROM in nested select in WHERE part
[info] - Find FROM in nested select (with multiple nested selects)
[info] Run completed in 480 milliseconds.
[info] Total number of tests run: 25
[info] Suites: completed 4, aborted 0
[info] Tests: succeeded 25, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] Error during tests:
[error] 	com.neowit.apex.SoqlGrammar
[error] 	com.neowit.apex.CodeCompletions
[info] Including from cache: commons-logging-1.1.3.jar
[info] Including from cache: apex-34.0.jar
[info] Including from cache: config-1.0.0.jar
[info] Including from cache: spray-json_2.10-1.3.0.jar
[info] Including from cache: force-wsc-34.2.2-uber.jar
[info] Including from cache: antlr-4.5-complete.jar
[info] Including from cache: tooling-34.0.jar
[info] Including from cache: partner-34.0.jar
[info] Including from cache: akka-actor_2.10-2.2-M1.jar
[info] Including from cache: metatdata-34.0.jar
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 4 s, completed Aug 2, 2015 5:14:10 AM
````
Stack trace
````
>  last test:test
[debug] Running TaskDef(com.neowit.apex.SoqlParserUtilsTest, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
[debug] Running TaskDef(com.neowit.apex.CodeCompletions, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
[debug] Running TaskDef(com.neowit.apex.SoqlGrammar, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
[debug] Running TaskDef(com.neowit.apex.metadata.StubFileGeneratorTest, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
[debug] Running TaskDef(com.neowit.apex.metadata.MetaXmlTest, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
[debug] Running TaskDef(com.neowit.apex.ConfigTest, org.scalatest.tools.Framework$$anon$1@3bbf0c63, false, [SuiteSelector])
java.lang.NullPointerException
	at com.neowit.apex.CodeCompletions.<init>(CodeCompletions.scala:17)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:621)
	at sbt.TestRunner.runTest$1(TestFramework.scala:76)
	at sbt.TestRunner.run(TestFramework.scala:85)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFunction.apply(TestFramework.scala:207)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
java.lang.NullPointerException
	at com.neowit.apex.SoqlGrammar.<init>(SoqlGrammar.scala:13)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at java.lang.Class.newInstance(Class.java:442)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:621)
	at sbt.TestRunner.runTest$1(TestFramework.scala:76)
	at sbt.TestRunner.run(TestFramework.scala:85)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFunction.apply(TestFramework.scala:207)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:226)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:235)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:226)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] Could not run test com.neowit.apex.SoqlGrammar: java.lang.NullPointerException
[error] Could not run test com.neowit.apex.CodeCompletions: java.lang.NullPointerException
[info] Run completed in 480 milliseconds.
[info] Total number of tests run: 25
[info] Suites: completed 4, aborted 0
[info] Tests: succeeded 25, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[debug] Passed tests:
[debug] 	com.neowit.apex.ConfigTest
[debug] 	com.neowit.apex.metadata.MetaXmlTest
[debug] 	com.neowit.apex.SoqlParserUtilsTest
[debug] 	com.neowit.apex.metadata.StubFileGeneratorTest
[error] Error during tests:
[error] 	com.neowit.apex.SoqlGrammar
[error] 	com.neowit.apex.CodeCompletions
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
````

**Comment by neowit** [comment]
same as for another issue #8 you opened - I have updated ["build"](https://github.com/neowit/tooling-force.com#building-tooling-forcecom) section in Readme. See if clarified build process will help you.

**Comment by kalyanlanka** [comment]
Thank you, it works.

**Comment by kalyanlanka** [comment]
Reopening this issue.  because while the problem with the build is addressed, the generated jar still does not work when running from command line whereas; the release package is working.  I think the build instructions need to be updated in [README](https://github.com/neowit/tooling-force.com/blob/master/README.md) to make sure they generate the same package as release.  Here is the error I get when I run the tooling-force.com-assembly-0.1-SNAPSHOT.jar.  No error when I run tooling-force.com-0.3.4.0 from [releases page](https://github.com/neowit/tooling-force.com/releases)

''''
Exception in thread "main" java.lang.NoClassDefFoundError: scala/runtime/Nothing$
	at com.neowit.apex.Runner.main(Runner.scala)
Caused by: java.lang.ClassNotFoundException: scala.runtime.Nothing$
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 1 more
''''

**Comment by kalyanlanka** [comment]
I get the above error if I run using java -jar 

but if I use scala here is the following error
````
scala tooling-force.com-assembly-0.1-SNAPSHOT.jar --help=bulkRetrieve
[ERROR] Executor - java.lang.NoSuchMethodError: scala.collection.immutable.$colon$colon.hd$1()Ljava/lang/Object;
java.lang.NoSuchMethodError: scala.Predef$.exceptionWrapper(Ljava/lang/Throwable;)Lscala/runtime/RichException;
	at com.neowit.apex.Executor.execute(Runner.scala:85)
	at com.neowit.apex.Runner$.main(Runner.scala:30)
	at com.neowit.apex.Runner.main(Runner.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at scala.reflect.internal.util.ScalaClassLoader$$anonfun$run$1.apply(ScalaClassLoader.scala:70)
	at scala.reflect.internal.util.ScalaClassLoader$class.asContext(ScalaClassLoader.scala:31)
	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.asContext(ScalaClassLoader.scala:101)
	at scala.reflect.internal.util.ScalaClassLoader$class.run(ScalaClassLoader.scala:70)
	at scala.reflect.internal.util.ScalaClassLoader$URLClassLoader.run(ScalaClassLoader.scala:101)
	at scala.tools.nsc.CommonRunner$class.run(ObjectRunner.scala:22)
	at scala.tools.nsc.JarRunner$.run(MainGenericRunner.scala:13)
	at scala.tools.nsc.CommonRunner$class.runAndCatch(ObjectRunner.scala:29)
	at scala.tools.nsc.JarRunner$.runJar(MainGenericRunner.scala:25)
	at scala.tools.nsc.MainGenericRunner.runTarget$1(MainGenericRunner.scala:69)
	at scala.tools.nsc.MainGenericRunner.run$1(MainGenericRunner.scala:87)
	at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:98)
	at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:103)
	at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
````

**Comment by kalyanlanka** [comment]
I think I resolved this issue. The issue seems to be with scala version.  The currently version of app, works only with 2.10 version.  My machine had 2.11 version

For that I had to upgrade my build.sbt and added required dependencies as XML and JSON parsers are deprecated 2.11

I think the README should specify the scala version.  

Ideally it would be better if there are different settings in build.sbt for 2.10 and 2.11 version.
